tests/unit/s2n_connection_serialize_test.c

```diff
@@ -473,6 +490,271 @@ int main(int argc, char **argv)
+
+            EXPECT_SUCCESS(s2n_negotiate_test_server_and_client(server_conn, client_conn));
+            EXPECT_EQUAL(s2n_connection_get_actual_protocol_version(server_conn), S2N_TLS13);
+
+            /* Get the client to do a read to pick up the session ticket. We won't use this one;
+             * we're interested in the tickets the server sends after the transfer. */
+            EXPECT_OK(s2n_send_and_recv_test(server_conn, client_conn));
+            EXPECT_SUCCESS(s2n_stuffer_wipe(&cb_session_data));
+
+            /* Serialize the server */
+            uint8_t buffer[S2N_SERIALIZED_CONN_TLS12_SIZE] = { 0 };
+            EXPECT_SUCCESS(s2n_connection_serialize(server_conn, buffer, sizeof(buffer)));
+
+            /* Initialize new server connection and deserialize session */
+            DEFER_CLEANUP(struct s2n_connection *new_server_conn = s2n_connection_new(S2N_SERVER),
+                    s2n_connection_ptr_free);
+            EXPECT_NOT_NULL(new_server_conn);
+            EXPECT_SUCCESS(s2n_connection_deserialize(new_server_conn, buffer, sizeof(buffer)));
+
+            /* Re-initialize IO pipes */
+            EXPECT_OK(s2n_connections_set_io_stuffer_pair(client_conn, new_server_conn, &io_pair));
+
+            /* The server automatically sends a ticket right after the handshake completes. However,
+             * we're interested in what happens after the server has been deserialized. Therefore we
+             * add another ticket for the server to send. */
+            EXPECT_SUCCESS(s2n_connection_add_new_tickets_to_send(new_server_conn, 1));
+
+            /* We want the server to be able to send a ticket so we reset the config */
+            EXPECT_SUCCESS(s2n_connection_set_config(new_server_conn, resumption_config));
+
+            /* Get the client to do a read to pick up the session ticket */
+            EXPECT_OK(s2n_send_and_recv_test(new_server_conn, client_conn));
+
+            /* Client should have received ST */
+            EXPECT_NOT_EQUAL(s2n_stuffer_data_available(&cb_session_data), 0);
+
+            /* Lets do a resumption handshake with the acquired ticket */
+            EXPECT_SUCCESS(s2n_shutdown_test_server_and_client(new_server_conn, client_conn));
+            EXPECT_SUCCESS(s2n_connection_wipe(client_conn));
+            EXPECT_SUCCESS(s2n_connection_wipe(new_server_conn));
+            EXPECT_OK(s2n_connections_set_io_stuffer_pair(client_conn, new_server_conn, &io_pair));
+
+            /* Client sets up a resumption connection with the received session ticket data */
+            size_t cb_session_data_len = s2n_stuffer_data_available(&cb_session_data);
+            EXPECT_SUCCESS(s2n_connection_set_session(client_conn, cb_session_data.blob.data, cb_session_data_len));
+            EXPECT_SUCCESS(s2n_stuffer_rewrite(&cb_session_data));
+
+            /* Negotiate successful resumption connection */
+            EXPECT_SUCCESS(s2n_negotiate_test_server_and_client(new_server_conn, client_conn));
```

Comment:
Should the client also be able to resume with the deserialized connection with a session ticket sent by the original server connection?

Reply:
Yeah, I added a test for this.